Library/Homebrew/os/linux/elf.rb

```diff
@@ -68,14 +97,19 @@ def binary_executable?
     elf_type == :executable
   end
 
+  def rpath
+    return @rpath if defined? @rpath
+
+    @rpath = HOMEBREW_PATCHELF_RB ? rpath_using_patchelf_rb : rpath_using_patchelf
```

Comment:
```suggestion
    @rpath ||= if HOMEBREW_PATCHELF_RB
      rpath_using_patchelf_rb
    else
      rpath_using_patchelf
    end
```

Reply:
This memoization does not work when `rpath` is `nil`.